lib/engine/step/dividend.rb

```diff
@@ -142,12 +142,21 @@ def change_share_price(entity, payout)
         return unless payout[:share_direction]
 
         prev = entity.share_price.price
-        payout[:share_times].times do
-          case payout[:share_direction]
-          when :left
-            @game.stock_market.move_left(entity)
-          when :right
-            @game.stock_market.move_right(entity)
+
+        Array(payout[:share_times]).each_with_index do |share_times, share_times_index|
```

Comment:
what about Array(times).zip(Array(direction)).each do |times, direction)

Reply:
I'll check out the zip method.

Reply:
Yep, that's a bit cleaner, thanks!